rust/kernel/types.rs

```diff
@@ -37,9 +36,22 @@ impl Mode {
 /// in kernel data structures, for example, an implementation of [`FileOperations`] in `struct
 /// file::private_data`.
 pub trait PointerWrapper {
+    /// Type of values borrowed between calls to [`PointerWrapper::into_pointer`] and
+    /// [`PointerWrapper::from_pointer`].
+    type Borrowed: Deref;
+
     /// Returns the raw pointer.
     fn into_pointer(self) -> *const c_types::c_void;
 
+    /// Returns a borrowed value.
+    ///
+    /// # Safety
+    ///
+    /// `ptr` must have been returned by a previous call to [`PointerWrapper::into_pointer`].
+    /// Additionally, [`PointerWrapper::from_pointer`] can only be called after *all* values
+    /// returned by [`PointerWrapper::borrow`] have been dropped.
```

Comment:
Are there a couple of safety constraints we've omitted here?
Following these `# Safety` instructions, are we allowed to:
- call `from_pointer()` multiple times on the same pointer?
- call `borrow()` after the call to `from_pointer()` ?

If so, does this merit a follow-up PR?

Reply:
Yes, this should be clarified. I'll prepare a follow-up PR.